doc/source/ray-core/handling-dependencies.rst

```diff
@@ -444,14 +444,95 @@ Caching and Garbage Collection
 """"""""""""""""""""""""""""""
 Runtime environment resources on each node (such as conda environments, pip packages, or downloaded ``working_dir`` or ``py_modules`` directories) will be cached on the cluster to enable quick reuse across different runtime environments within a job.  Each field (``working_dir``, ``py_modules``, etc.) has its own cache whose size defaults to 10 GB.  To change this default, you may set the environment variable ``RAY_RUNTIME_ENV_<field>_CACHE_SIZE_GB`` on each node in your cluster before starting Ray e.g. ``export RAY_RUNTIME_ENV_WORKING_DIR_CACHE_SIZE_GB=1.5``.
 
-When the cache size limit is exceeded, resources not currently used by any actor, task or job will be deleted.
+When the cache size limit is exceeded, resources not currently used by any Actor, Task or Job are deleted.
+
+.. runtime-environments-job-conflict:
+
+Runtime Environment Specified by Both Job and Driver
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+You can specify Ray driver's runtime environment via `ray.init(runtime_env=...)` or `ray job submit --runtime-env`.
```

Comment:
Ray driver's runtime env can only be specified via job submission. ray.init runtime env has no effect on the driver.

Reply:
Oh great catch.  `ray.init()` runtime env only sets the default for any subsequent task or actor's runtime env

Reply:
I think actually ray job submit also is not for driver's runtime env? Does it mean ray job submit creates a runtime env before it submits a driver?

Reply:
IIUC, ray job submit just passes the config via env var, and both will just set runtime env at ray.init which is used for actor/task?

Reply:
wait, so if we do job submission, the driver script will already have a runtime env created? How does it work now? Is the runtime env created by supervisor actor first and the driver is started with the same config?

Reply:
It looks like it is already written in a pretty confusing way; https://docs.ray.io/en/latest/ray-core/handling-dependencies.html#specifying-a-runtime-environment-per-job (it considers as job submit & ray.init as "runtime env per job"). I think we should technically distinguish them in this case. Let me try rewriting this part

Reply:
> Is the runtime env created by supervisor actor first and the driver is started with the same config?

That's right, the driver is started by the actor as a subprocess so it inherits the cwd and env vars automatically without Ray having to do anything.